lib/chef/run_list/run_list_expansion.rb

```diff
@@ -140,8 +152,9 @@ def applied_role(role_name)
       end
 
       def expand_run_list_items(items, included_by="top level")
+
         if entry = items.shift
-          @run_list_trace[included_by.to_s] << entry.to_s
+          @run_list_trace[included_by.to_s] <<  entry
```

Comment:
This line does potentially change a visible API external to this object.  Someone could be calling `#run_list_trace` to get back this hash and be expecting string arguments.